File: Leetcode/2181 Merge Nodes in Between Zeros.py
class Solution:
    def mergeNodes(self, head):
      left, right = head, head.next
      while right is not None:
        if right.val == 0:
          left.next = None if right.next is None else right
          left = right
        else:
          left.val += right.val
        right = right.next
      return head
    

File: Leetcode/2181 Merge Nodes in Between Zeros.ts
function mergeNodes(head: ListNode){
  let left = head, right= head.next;
  while (right !== null) {
    if (right.val === 0) {
      left.next = right.next === null ? null : right;
      left = right;
    } else {
      left.val += right.val; 
    }
    right = right.next;
  }
  return head;
}
